File: src/1614-Maximum-Nesting-Depth-of-the-Parentheses/maximum_nesting_depth_of_the_parentheses.cpp
#include <iostream>
#include <algorithm>
#include <string>

class Solution {
public:
    int maxDepth(std::string s) {
        int max_depth = 0;
        int current_depth = 0;
        
        for (char& c : s) {
            if (c == '(') {
                current_depth += 1;
                max_depth = std::max(max_depth, current_depth);
            } else if (c == ')') {
                current_depth -= 1;
            }
        }
        
        return max_depth;
    }
};

int main() {
    std::string s = "(1+(2*3)+((8)/4))+1";
    
    Solution solution;
    
    std::cout << std::boolalpha << solution.maxDepth(s) << std::endl;
    
    return 0;
}

File: src/1614-Maximum-Nesting-Depth-of-the-Parentheses/maximum_nesting_depth_of_the_parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max_depth: int = 0
        current_depth : int = 0
        
        for char in s:
            if char == '(':
                current_depth += 1
                max_depth = max(max_depth, current_depth)
            elif char == ')':
                current_depth -= 1
                
        return max_depth
    
if __name__ == "__main__":
    s: str = "(1+(2*3)+((8)/4))+1"
    solution = Solution()
    
    print(solution.maxDepth(s))